test/shell/test_scala_binary.sh

```diff
@@ -18,5 +18,26 @@ test_scala_binary_expect_failure_on_missing_direct_deps_located_in_dependency_wh
   test_scala_library_expect_failure_on_missing_direct_deps ${dependency_target} ${test_target}
 }
 
+test_scala_binary_allows_opt_in_to_use_of_argument_file_in_runner_for_improved_performance() {
+
+  bazel run --extra_toolchains="//test/toolchains:use_argument_file_in_runner" //test/src/main/scala/scalarules/test/large_classpath:largeClasspath
+  grep "\"True\" == \"True\"" bazel-bin/test/src/main/scala/scalarules/test/large_classpath/largeClasspath
```

Comment:
Hi @ittaiz, overall looks good to me. I just find this grep pattern non intuitive. Maybe template could render something like `test_runner_classpath_mode="argsfile"|"manifest"` do an if on that variable and then grep for `test_runner_classpath_mode="argsfile"` in test? wdyt?

Reply:
sure. I can do that. I also didn't like the grep. Vaidas wanted the `use_argument_file` thing and I went with it all the way. Will do the change.
To make sure we're aligned. 
The new greps will be:
1. `"argsfile" == "argsfile"` 
2. `"manifest" == "argsfile"`
Right?

Reply:
@simuons PTAL